bson/src/main/org/bson/Document.java

```diff
@@ -141,6 +141,24 @@ public Document append(final String key, final Object value) {
     }
 
     /**
+     * Gets the value of the given key, casting it to the given {@code Class<T>} or returns the default value if null.
+     * This is useful to avoid having casts in client code, though the effect is the same.
+     *
+     * @param key   the key
+     * @param defaultValue what to return if the value is null
+     * @param clazz the non-null class to cast the value to
+     * @param <T>   the type of the class
+     * @return the value of the given key, or null if the instance does not contain this key.
+     * @throws ClassCastException if the value of the given key is not of type T
+     * @since 3.5
+     */
+    public <T> T get(final Object key, final T defaultValue, final Class<T> clazz) {
```

Comment:
Sorry, but I had another think about this method, and am wondering whether we can simplify it and at the same time not need all the more type-specific methods added below.  If defaultValue is required to be non-null, then clazz is redundant, and we just have:

    public <T> T get(final Object key, final T defaultValue) {
        notNull("defaultValue", defaultValue);
        notNull("clazz", (Class<T>) defaultValue.getClass());
        Object value = documentAsMap.get(key);
        return value == null ? defaultValue : ((Class<T>) defaultValue.getClass()).cast(value);
    }

and the type-specific methods are not necessary, as it's just as easy to do:

     long val = doc.get("f1", 42L);
     Date dVal = doc.get("f2", new Date());

The only downside I see is that there is already an overload of this method signature:

    public <T> T get(final Object key, final Class<T> clazz) ...

but they are unlikely to cause ambiguity in common usage.  

It's also a bit friendlier for users of a non-default BsonTypeClassMap, e.g.

    Instant val = doc.get("f2", Instant.now());